windows-server-container-tools/CleanupContainerHostNetworking/README.md

```diff
@@ -1,43 +1,58 @@
+- Stop/Remove all containers on the host, regardless of their state 
+- Remove all container networks on the host
 
-```none
-PS C:\> .\CleanupContainerHostNetworking.ps1 -ForceCleanup
+In addition to the basic `-Cleanup` option, there is also the `-ForceDeleteAllSwitches` option. *Use these options together to force an extended host cleanup*, in which Switch/NIC registry keys are removed from your host, network adapters are unbound and the host default NAT network is removed, and the HNS.data file is deleted to remove all current HNS configurations.
+```
+PS C:\> .\WindowsContainerNetworking-LoggingAndCleanupAide.ps1 -Cleanup -ForceDeleteAllSwitches
 ```
+## Assumptions/System Requirements - *READ BEFORE RUNNING SCRIPT*
+
+**WARNING:** This script requires that the host machine is not running in swarm mode. The script will make sure the host is in an inactive swarm state. If the host is in swarm mode, the script will provide an option to leave swarm mode.
+
+This script assumes the following about your system. Ensure these requirements are met before running this script on your system:
+- Docker (Docker CE, Docker EE, or Docker for Windows) is installed *and running* 
+- Microsoft Hyper-V role is enabled 
+- The Windows Containers feature is installed
+
+## \*Info/logs collected for capturing container host state
+*Whether it is run without arguments, or with any of the arguments above, as part of its basic logging functionality this script captures:*
+- The Host Network Service (HNS) event log (`c:\Windows\System32\winevt\Logs\Microsoft-Windows-Host-Network-Service-Admin.evtx`)
+- The System event log (`c:\Windows\System32\winevt\Logs\System.evtx`)
+- The HNS.data file (`c:\ProgramData\Microsoft\Windows\HNS\HNS.data`)
+- Information on the [Virtual Filtering Platform (VFP)](https://www.microsoft.com/en-us/research/project/azure-virtual-filtering-platform/), virtual switch extension which is running as a service on the host (`PS C:\> Get-Service vfpext`)
+- VFP policy information for all virtual switches 
+- Docker version (`C:\> docker -v`)
+- Docker info (`C:\> docker info`)
+- Information on current TCP/IP, DHCP and DNS network configuration values on the host (`C:\> ipconfig /allcompartments /all`)
+- The virtual switches on this host (`PS C:\> Get-VMSwitch`)
+- The NAT objects on this host (`PS C:\> Get-NetNat`)
+- The static mappings configured on the host's NAT instances (`PS C:\> Get-NetNatStaticMapping`)
+- The network adapters on this host (`PS C:\> Get-NetAdapter -IncludeHidden`)
+- The bindings for the network adapters on this host (`PS C:\> Get-NetAdapterBinding -IncludeHidden`)
+- The status of the Windows NAT driver (`PS C:\> Get-Service winnat`)
+- The status of the Windows Firewall Service (`PS C:\> Get-Service mpssvc`)
+- The HNS registry settings (`PS C:\> Get-ChildItem HKLM:\SYSTEM\CurrentControlSet\Services\hns`)
+- The Hyper-V registry settings (`PS C:\> Get-ChildItem HKLM:\SYSTEM\CurrentControlSet\Services\vmsmp`)
```

Comment:
Done. (I had the prompts to make it clearer when commands were PowerShell commands, but I guess people can just look that up if they need to).